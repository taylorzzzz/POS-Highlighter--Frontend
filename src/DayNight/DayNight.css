.DayNight {

}

.DayNight label {
    outline: 0;
    background: var(--color-disabled);
    display: block;
    width: 3em;
    height: 2em;
    position: relative;
    cursor: pointer;
    user-select: none;
    border-radius: 2em;
    padding: 5px;
    transition: all .4s ease;
}
.DayNight label:before,
.DayNight label:after {
    position: relative;
    display: block;
    content: "";
    width: 50%;
    height: 100%;
}

.DayNight label:after {
    left: 0;
    border-radius: 50%;
    background: var(--color-text);
    transition: all .2s ease;
    
}
.DayNight label:before {
    display: none;
}

#toggle-btn:checked + label {
    background: var(--color-text);
}
#toggle-btn:checked + label:after {
    left: 50%;
    background: var(--color-bg);
}

#toggle-btn {
    display: none;
}

