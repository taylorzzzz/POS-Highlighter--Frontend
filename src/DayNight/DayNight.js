import React from 'react';

import './DayNight.css';

function DayNight(props) {

    return  (
        <div className="DayNight">
            <input 
                type="checkbox" 
                checked={props.dayToggled}
                id="toggle-btn" 
                onChange={props.toggleDayNight}></input>
            <label htmlFor="toggle-btn"></label>
        </div>
    )
}

export default DayNight;