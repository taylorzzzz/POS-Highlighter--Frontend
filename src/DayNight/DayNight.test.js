import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import DayNight from './DayNight';

test('DayNight renders properly', () => {
    const tree = renderer.create(<DayNight />).toJSON();
    
    expect(tree).toMatchSnapshot();
})

test('DayNight toggleDayNight events properly', () => {

    const spyToggle = jest.fn();

    const wrapper = shallow(<DayNight toggleDayNight={spyToggle}/>);

    expect(spyToggle).toHaveBeenCalledTimes(0);

    const input = wrapper.find('input');
    input.simulate('change');

    expect(spyToggle).toHaveBeenCalled();
    
})

test('DayNight dayToggled prop gets passed to checked properly', () => {

    const initialVal = false;

    const wrapper = shallow(<DayNight dayToggled={initialVal}/>);

    const input = wrapper.find('input');

    expect(input.props().checked).toBe(initialVal);
    
})