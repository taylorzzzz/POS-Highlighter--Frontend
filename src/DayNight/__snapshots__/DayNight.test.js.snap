// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayNight renders properly 1`] = `
<div
  className="DayNight"
>
  <input
    checked={undefined}
    id="toggle-btn"
    onChange={undefined}
    type="checkbox"
  />
  <label
    htmlFor="toggle-btn"
  />
</div>
`;
